pkg/lint/rules/template.go

```diff
@@ -20,14 +20,15 @@ import (
 	"bufio"
 	"bytes"
 	"fmt"
+	"io"
 	"os"
 	"path"
 	"path/filepath"
 	"regexp"
 	"strings"
 
 	"github.com/pkg/errors"
-	"sigs.k8s.io/yaml"
+	"k8s.io/apimachinery/pkg/util/yaml"
```

Comment:
this might be considered a breaking change as it's unclear whether these two yaml parsers are compatible with each other. One YAML parser's output can differ from another.

Can you see if this can be reverted?

Reply:
The main feature why I have changed to `apimachinery` is that it supports multi-document YAML files correctly (the `---` separator).

Just checked which YAML parser is used underneath, and `apimachinery` uses the same: `sigs.k8s.io/yaml`, it just wraps it to add multi-document support, so I think it is safe to say that it is the same YAML parser.

https://github.com/kubernetes/apimachinery/blob/d10b9c3a6d2ee7cd58bec9931482a0d604534231/pkg/util/yaml/decoder.go#L32

Reply:
Looked at the code in `k8s.io/apimachinery/pkg/util/yaml` and also discussed with @mattfarina. The code wraps `sigs.k8s.io/yaml` with an IO reader to read multi-doc yaml file and then calls the yaml unmarshall for each doc contained. It is using the same parser as previous and good to use.